include/cglm/types.h

```diff
@@ -48,8 +48,17 @@ typedef CGLM_ALIGN_IF(16) vec4  mat4[4];
 
 typedef vec4                    versor;
 
-#define CGLM_PI    ((float)M_PI)
-#define CGLM_PI_2  ((float)M_PI_2)
-#define CGLM_PI_4  ((float)M_PI_4)
+#define GLM_PI     3.14159265358979323846264338327950288
+#define GLM_PI_2   1.57079632679489661923132169163975144
+#define GLM_PI_4   0.785398163397448309615660845819875721
```

Comment:
Since we defined math constants **manually** and since it might not be defined in anywhere else maybe defining other numbers would also help developers who use them. 

I think defining other numbers here is not bad idea. Other definitions:

```C
#define GLM_E         2.71828182845904523536028747135266250   /* e              */
#define GLM_LOG2E     1.44269504088896340735992468100189214   /* log2(e)        */
#define GLM_LOG10E    0.434294481903251827651128918916605082  /* log10(e)       */
#define GLM_LN2       0.693147180559945309417232121458176568  /* loge(2)        */
#define GLM_LN10      2.30258509299404568401799145468436421   /* loge(10)       */
#define GLM_PI        3.14159265358979323846264338327950288   /* pi             */
#define GLM_PI_2      1.57079632679489661923132169163975144   /* pi/2           */
#define GLM_PI_4      0.785398163397448309615660845819875721  /* pi/4           */
#define GLM_1_PI      0.318309886183790671537767526745028724  /* 1/pi           */
#define GLM_2_PI      0.636619772367581343075535053490057448  /* 2/pi           */
#define GLM_2_SQRTPI  1.12837916709551257389615890312154517   /* 2/sqrt(pi)     */
#define GLM_SQRT2     1.41421356237309504880168872420969808   /* sqrt(2)        */
#define GLM_SQRT1_2   0.707106781186547524400844362104849039  /* 1/sqrt(2)      */
```

I borrowed this in `math.h` header in macOS and added `GL` prefix to them. We can use them with their comments.

If using all is redundant for maintenance then we could pick related numbers like 1/PI, 2/PI... What do you think? 

It is not required in this PR, I can add this later if you want.

Reply:
Whoops, missed those somehow. Fixed that now.

I guess I'll just add those numbers as well, while we're at it. None of those `M_` constants is in standard C, so it makes a lot of sense for a math library like this to provide them.